src/components/log-in/LogInForm.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button, Form, Icon, Input } from "antd";
+import * as React from "react";
+
+const FormItem = Form.Item;
+
+function hasErrors(fieldsError: any) {
+    return Object.keys(fieldsError).some((field) => fieldsError[field]);
+}
+
+export class LoginForm extends React.Component<any, any> {
+
+    public componentDidMount() {
+        // To disabled submit button at the beginning.
+        this.props.form.validateFields();
+    }
+
+    public handleSubmit = (e: any) => {
+        e.preventDefault();
+        this.props.form.validateFields((err: any, values: any) => {
```

Comment:
see if type of `err` can be `Error` and for values you can define the interface. since from your form you are getting two values: `userName` and `password`, you can define an interface : 

    interface LoginData {
      userName: string;
      password: string;
    }